vm/src/stdlib/itertools.rs

```diff
@@ -181,6 +184,141 @@ impl PyItertoolsTakewhile {
     }
 }
 
+#[pyclass(name = "islice")]
+#[derive(Debug)]
+struct PyItertoolsIslice {
+    iterable: PyObjectRef,
+    cur: RefCell<usize>,
+    next: RefCell<usize>,
+    stop: Option<usize>,
+    step: usize,
+}
+
+impl PyValue for PyItertoolsIslice {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("itertools", "islice")
+    }
+}
+
+fn pyobject_to_usize(obj: PyObjectRef, vm: &VirtualMachine) -> Option<usize> {
+    let is_int = objtype::isinstance(&obj, &vm.ctx.int_type());
+    if is_int {
+        objint::get_value(&obj).to_usize()
+    } else {
+        None
+    }
+}
+
+#[pyimpl]
+impl PyItertoolsIslice {
+    #[pymethod(name = "__new__")]
+    fn new(_cls: PyClassRef, args: PyFuncArgs, vm: &VirtualMachine) -> PyResult {
```

Comment:
Try something like:

```rust
fn new(_cls: PyClassRef, iter: PyIterable, start: Option<PyInt>, stop:  OptionalArg<Option<PyInt>>, step: OptionalArg<Option<PyInt>>, vm: &VirtualMachine) -> PyResult {
    let (start, stop, step) = match (start, stop.into_option().and_then(|x| x), step.into_option().and_then(|x| x)) {
        ...
    };
}
```

Reply:
Actually I don't think I can use it because it then when calling `islice(x, 5)` then `5` is treated as the `start` argument - not as the `stop` one, as required by CPython.

Reply:
It's indeed nicer to declare the arguments in the function prototype, but in this case it's not much of a a help because the argument order changes depending on the number of them present.